daml/Marketplace/Derivative.daml

```diff
@@ -0,0 +1,127 @@
+module Marketplace.Derivative where
+
+import DA.Finance.Types
+
+import DA.Next.Set
+
+
+data SecType = CMDTYSWAP | FRA | FUT | FWD | IRS | MLEG | OPT | SWAPTION
+  deriving (Eq, Show)
+
+
+data ExerciseStyle = European | American | Bermudan
+  deriving (Eq, Show)
+
+
+data PutOrCall = Call | Put
+  deriving (Eq, Show)
+
+
+data OptionData = OptionData
+  with
+    putCall : PutOrCall  -- put or call option
+    strikePx : Decimal  -- the Strike Price
+    exerStyle : ExerciseStyle  -- option exercise style
+  deriving (Eq, Show)
+
+
+template Derivative
+  with
+    id : Id
+    pricePrecision : Int
+    description : Text
+    prodSym : Text  -- BTCF21
+    prodCode : Text  -- BTC
+    uuid : Text
+    secType : SecType
+    mmy : Text  -- YYYYMM or YYYYMMDD or YYYMMwN
```

Comment:
why not a `Date`?

Reply:
Dimitri added this, but I'm pretty sure it's the specified date format as opposed to the actual date

Reply:
Oh, I see. We might want to make it an enum or something if that's the case, but this field also doesn't seem to be in use anywhere, so we can leave it be as well

Reply:
Yeah agreed. I'll do some digging to see if it will be used and how, but we'll leave it for now.